vision_processing/vision/reference_point.py

```diff
@@ -22,44 +22,48 @@ def __init__(self, pose_to_robot: Pose, pose_to_field: Pose):
         self.robot_pose = robot_to_field
 
     @classmethod
-    def from_apriltags(cls, camera: Camera) -> list["ReferencePoint"]:
+    def from_apriltags(cls, camera: Camera) -> ReferencePoint | None:
         """
         Create a List of ReferencePoint from an image
         :rtype ReferencePoint
         """
         detector = apriltags.Detector(GameField.apriltag_family)
         image = cv2.cvtColor(camera.frame, cv2.COLOR_BGR2GRAY)
 
-        # noinspection PyTypeChecker
-        detections = detector.detect(
-            image,
-            estimate_tag_pose=True,
-            camera_params=(
-                camera.center_pixel_height,
-                camera.center_pixel_height,
-                camera.center.x,
-                camera.center.y,
-            ),
-            tag_size=GameField.apriltag_size,
-        )
+        best_detection = None
+        best_detection_decision_margin = 10
 
-        reference_points = [
-            cls(
+        # noinspection PyTypeChecker
+        for detection in detector.detect(
+                image,
+                estimate_tag_pose=True,
+                camera_params=(
+                        camera.center_pixel_height,
+                        camera.center_pixel_height,
+                        camera.center.x,
+                        camera.center.y,
+                ),
+                tag_size=GameField.apriltag_size,
+        ):
+            if detection.decision_margin > best_detection_decision_margin:
```

Comment:
It looks like you're trying to select the detection with the highest decision margin from the array returned on line 37? If that's the case I'd recommend using `max` and specifying the key for comparison. Here's an article that shows how to do that: https://www.geeksforgeeks.org/python-get-the-object-with-the-max-attribute-value-in-a-list-of-objects/